docs/operational_guide/resource_limits.md

```diff
@@ -4,16 +4,93 @@ This operational guide provides an overview of how to set resource limits on
 M3 components to prevent abusive reads/writes impacting availability or 
 performance of M3 in a production environment.
 
+## M3 Query and M3 Coordinator
+
+### Deployment
+
+Protecting queries impacting your ingestion of metrics for metrics workloads 
+can first and foremost be done by deploying M3 Query and M3 Coordinator 
+independently. That is, for writes to M3 use a dedicated deployment of 
+M3 Coordinator instances, and then for queries to m3 use a dedicated deployment 
+of M3 Query instances.
+
+This ensures when M3 Query instances become busy and are starved of resources 
+serving an unexpected query load, they will not interrupt the flow of metrics
+being ingested to M3.
+
+### Configuring limits
+
+To protect individual queries using too many resources, you can specify some
+sane limits in the M3 Query (and consequently M3 Coordinator) configuration 
+file under the top level `limits` config stanza.
+
+There are two types of limits:
+
+- Per query time series limit
+- Per query time series * blocks limit (docs limit)
+
+When either of these limits are hit, you can define the behavior you would like, 
+either to return an error when this limit is hit, or to return a partial result 
+with the response header `M3-Results-Limited` detailing the limit that was hit 
+and a warning included in the response body.
+
+### Annotated configuration
+
+```
+limits:
+  # If set will override default limits set per query.
+  perQuery:
+    # If set limits the number of time series returned for any given 
+    # individual storage node per query, before returning result to query 
+    # service.
+    maxFetchedSeries: 0
+
+    # If set limits the number of index documents matched for any given 
+    # individual storage node per query, before returning result to query 
+    # service.
+    # This equates to the number of time series * number of blocks, so for 
+    # 100 time series matching 4 hours of data for a namespace using a 2 hour 
```

Comment:
Hm wouldn't this be `100 series * 4 hours / 2 hour block size` == `200 docs`?

Reply:
Ah yes, heh, bad maths. Thank you.